src/Storefront/Resources/app/storefront/src/component-system/README.md

```diff
@@ -0,0 +1,361 @@
+
+To define which elements the component should be applied to, you can define a static property `selector` in your component class. The component will then automatically be initialized on elements that match the selector. This also applies to elements that might be added later to the document. You do not need to do this manually.
+
+```javascript
+class MyComponent extends ShopwareComponent {
+
+    static selector = '[data-my-component]';
+    
+    init() {
+        console.log('Component automatically initialized!');
+    }
+}
+```
+
+## Component Configuration
+
+### Data Attributes
+
+Components can be configured through data attributes using the pattern `data-{component-name}-options`. For example, you can pass information form Twig into your component. The options should be passed as a JSON string.
+
+```Twig
+
+{% set componentOptions = {
+    foo: "bar" 
+    test: true
+} %}
+
+<div data-my-component 
+     data-my-component-options="{{ componentOptions|json_encode }}">
+</div>
+```
+
+The component name is automatically converted to dash-case for the data attribute. The passed options are merged with the default options that you define as a static property in your component class.
+
+## Directory Structure & Component Script Loading
+
+Component scripts are automatically loaded when the corresponding Twig component is used within the page. Your component script should have the same name as the Twig file of your component and should be placed in the same directory. Shopware will automatically collect all component script files and include them into the page if the component is used on a specific page.
+
+Example structure:
+
+```
+views/
+  components/
+    MyComponentNamespace/
+      MyComponent.thml.twig
+      MyComopnent.js
+      MyComponent.scss
+```
+
+## Twig Component Integration
+
+To integrate the script with your corresponding Twig component you have to ensure that the desired element within your component template has the necessary data attributes.
+
+You can build the options individually from your Twig component properties and other data, or simply pass through the component properties to your JavaScript component.
+
+```twig
+{# views/component/MyComponent.html.twig #}
+
+{% props
+    foo = "bar",
+    custom = true,
+%}
+
+<div data-my-component 
+     data-my-component-options='{{ this.props|json_encode }}'>
+
+    {# Some component logic ... #}
+</div>
+```
+
+
+## Component Communication
+
+### Event System
+
+To react to actions from other components, there is a new central event system available which can be accessed via the global `window.Shopware` singleton.
+
+In your component you can emit events to inform others about an action and pass additional data via the event.
+
+```javascript
+// MyComponent.js
+
+class MyComponent extends ShopwareComponent {
+
+    // ...
+
+    doSomething() {
+        const message = 'Hello World!';
+
+        window.Shopware.emit('MyComponent:DoSomething', message);
```

Comment:
If we are emitting an event, and the event is prefixed with the component name, we could maybe also use `this.componentName`.

```suggestion
        window.Shopware.emit(`${this.componentName}:DoSomething`, message);
```